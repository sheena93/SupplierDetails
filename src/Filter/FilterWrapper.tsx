import React, { ReactNode } from 'react';
import { Box, Button, Divider, Drawer, Grid, TypeBase } from '@c2fo/components';

import { useStyles } from './FilterWrapper.style';
// import { T } from '../../../i18n';

const TESTID_ROOT = 'filterDrawer';
export const testIds = {
  drawer: `${TESTID_ROOT}:drawerComponent`,
  applyButton: `${TESTID_ROOT}:applyButton`,
  cancelButton: `${TESTID_ROOT}:cancelButton`,
  resetButton: `${TESTID_ROOT}:resetButton`,
} as const;

type FilterDrawerProps = {
  showDrawer: boolean;
  applyFilter: () => void;
  resetFilter: () => void;
  cancelFilter?: () => void;
};

export const FilterDrawer: React.FC<FilterDrawerProps & ReactNode> = ({
  children,
  showDrawer,
  applyFilter,
  resetFilter,
  cancelFilter,
}) => {
  const classes = useStyles();

  return (
    <Drawer
      data-testid={testIds.drawer}
      anchor="right"
      open={showDrawer}
      classes={{
        paper: classes.drawerPaper,
      }}
      ModalProps={
        cancelFilter
          ? {
              onBackdropClick: () => {
                cancelFilter();
              },
            }
          : {}
      }
    >
      <div className={classes.drawer}>
        <Box className={classes.addFiltersTitle}>
          <TypeBase isEmphasis>
            {/* todo add translation */}
            {/* <T k="core.addFilters">ADD FILTERS</T> */}
            ADD FILTERS
          </TypeBase>
        </Box>
        <Divider />
        <div className={classes.Drawerchildren}> {children} 
        <Button
          className={classes.resetFilters}
          data-testid={testIds.resetButton}
          variant="text"
          color="secondary"
          onClick={resetFilter}
        >
          {/* todo add translation */}
          {/* <T k="core.resetFilter">Reset Filters</T> */}
          Reset Filters
        </Button>
        </div>
        <Grid container className={classes.filterButtons} spacing={1}>
          <Grid item sm="auto" xs={12}>
            <Button
              data-testid={testIds.cancelButton}
              className={classes.cancelButton}
              size="large"
              variant="text"
              color="secondary"
              onClick={cancelFilter}
            >
              {/* <T k="core.cancel">Cancel</T> */}
              Cancel
            </Button>
          </Grid>
          <Grid item sm="auto" xs={12}>
            <Button
              data-testid={testIds.applyButton}
              className={classes.applyButton}
              size="large"
              variant="contained"
              onClick={applyFilter}
            >
              {/* <T k="core.apply">Apply</T> */}
              Apply
            </Button>
          </Grid>
        </Grid>
      </div>
    </Drawer>
  );
};