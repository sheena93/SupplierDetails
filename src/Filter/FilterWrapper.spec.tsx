import React from "react";
import { screen } from "@testing-library/dom";
import { FilterDrawerComponent } from "./FilterWrapper";
import { render, fireEvent, act } from "@testing-library/react";

const cancelFilter = jest.fn();
const applyFilter = jest.fn();
const resetFilter = jest.fn();

describe("<FilterDrawerComponent />", () => {
  it("should render with required props and show drawer true", async () => {
    render(
      <FilterDrawerComponent
        applyFilter={applyFilter}
        resetFilter={resetFilter}
        cancelFilter={cancelFilter}
        showDrawer={true}
        children
      />
    );
    expect(FilterDrawerComponent).toMatchSnapshot();
    
  });
  it("should render with required props and show drawer false and children <div>", async () => {
    render(
      <FilterDrawerComponent
        applyFilter={applyFilter}
        resetFilter={resetFilter}
        cancelFilter={cancelFilter}
        showDrawer={false}
        children ={<div></div>}
      />
    );
    expect(FilterDrawerComponent).toMatchSnapshot();
  });

  it("should render FilterDrawerComponent multiple children", async () => {
    await act(async () => {
      await render(
        <FilterDrawerComponent
          applyFilter={applyFilter}
          resetFilter={resetFilter}
          cancelFilter={cancelFilter}
          showDrawer={true}
          children = {[<div></div>,<div></div>]}
        />
      );
    });
  });

  it("should handle Apply click", async () => {
    await act(async () => {
      await render(
        <FilterDrawerComponent
          applyFilter={applyFilter}
          resetFilter={resetFilter}
          cancelFilter={cancelFilter}
          showDrawer={true}
          children
        />
      );
      const applyButton = screen.getByText("Apply");
      fireEvent.click(applyButton);
      expect(applyFilter).toBeCalled();
    });
  });

  it("should handle Reset click", async () => {
    await act(async () => {
      await render(
        <FilterDrawerComponent
          applyFilter={applyFilter}
          resetFilter={resetFilter}
          cancelFilter={cancelFilter}
          showDrawer={true}
          children
        />
      );
      const resetButton = screen.getByText("Reset Filters");
      fireEvent.click(resetButton);
      expect(resetFilter).toBeCalled();
    });
  });

  it("should handle Cancel click", async () => {
    await act(async () => {
      await render(
        <FilterDrawerComponent
          applyFilter={applyFilter}
          resetFilter={resetFilter}
          cancelFilter={cancelFilter}
          showDrawer={true}
          children
        />
      );
      const cancelbutton = screen.getByText("Cancel");
      fireEvent.click(cancelbutton);
      expect(cancelFilter).toBeCalled();
    });
  });

});
