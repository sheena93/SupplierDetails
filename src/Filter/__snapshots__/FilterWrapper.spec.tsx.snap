// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterDrawerComponent /> should render with required props 1`] = `[Function]`;

exports[`<FilterDrawerComponent /> should render with required props 2`] = `[Function]`;

exports[`<FilterDrawerComponent /> should render with required props and show drawer false 1`] = `[Function]`;

exports[`<FilterDrawerComponent /> should render with required props and show drawer true 1`] = `[Function]`;
